native/src/widget/scrollable.rs

```diff
@@ -35,9 +37,36 @@ impl<'a, Message, Renderer: self::Renderer> Scrollable<'a, Message, Renderer> {
             scroller_width: 10,
             content: Column::new(),
             style: Renderer::Style::default(),
+            on_scroll: None,
+            snap_to_bottom: false,
         }
     }
 
+    /// Set the [`Scrollable`] to snap to bottom when the user
+    /// scrolls to bottom. This will keep the scrollable at the bottom
+    /// even if new content is added to the scrollable.
+    ///
+    /// [`Scrollable`]: struct.Scrollable.html
+    pub fn snap_to_bottom(mut self) -> Self {
+        self.snap_to_bottom = true;
+        self
+    }
```

Comment:
I'm envisioning a chat app with a checkbox for "Scroll Automatically." With this pattern, the implementation needs to do a conditional builder call:

```rust
let mut scrollable = Scrollable::new();
if settings.snap {
    scrollable = scrollable.snap_to_bottom();
}
```

It might be nicer to simply pass a bool:

```rust
let scrollable = Scrollable::new()
    .snap_to_bottom(settings.snap);
```

Reply:
Yeah, that would indeed look nicer. And I suppose `snap_to_bottom(true / false)` shows the intent better than `snap_to_bottom()`. I'll change that.